lib/trino-parquet/src/main/java/io/trino/parquet/reader/ParquetReader.java

```diff
@@ -415,21 +412,20 @@ private ColumnChunk readStruct(GroupField field)
         return new ColumnChunk(rowBlock, columnChunk.getDefinitionLevels(), columnChunk.getRepetitionLevels());
     }
 
-    private FilteredOffsetIndex getFilteredOffsetIndex(int rowGroup, long rowGroupRowCount, ColumnPath columnPath)
+    @Nullable
+    private FilteredOffsetIndex getFilteredOffsetIndex(FilteredRowRanges rowRanges, int rowGroup, long rowGroupRowCount, ColumnPath columnPath)
     {
-        if (filter.isPresent()) {
-            RowRanges rowRanges = getRowRanges(filter.get(), rowGroup);
-            if (rowRanges != null && rowRanges.rowCount() < rowGroupRowCount) {
-                Optional<ColumnIndexStore> columnIndexStore = this.columnIndexStore.get(rowGroup);
-                if (columnIndexStore.isPresent()) {
-                    OffsetIndex offsetIndex = columnIndexStore.get().getOffsetIndex(columnPath);
-                    if (offsetIndex != null) {
-                        return FilteredOffsetIndex.filterOffsetIndex(offsetIndex, rowRanges, rowGroupRowCount);
-                    }
-                }
-            }
+        if (rowRanges == null) {
+            return null;
```

Comment:
It'd be cleaner for the caller to make this check and avoid calling this method altogether if rowRanges is null.

Reply:
I tried moving the null check to the two callers of this method and the resulting code wasn't cleaner

Reply:
It's cleaner from an abstraction perspective. The reader doesn't have to reason about why rowRanges might be null, especially in relationship with the corresponding rowGroup. In general, I think this method and the other related methods need to be restructured to have a clearer set of responsibilities (looking up the ColumnIndexStore, the OffsetIndex, etc)

Reply:
Updated to add the null check to callers and removed from here